clients/fides-js/src/lib/i18n/i18n-utils.ts

```diff
@@ -72,11 +131,28 @@ function extractMessagesFromExperienceConfig(
       // For each translation, extract each of the translated fields
       (translation: ExperienceConfigTranslation) => {
         const locale = translation.language;
+        // We only override experience translations if the override_language matches current locale
+        let localeHasOverride = false;
+        if (experienceTranslationOverrides?.override_language) {
+          localeHasOverride = !!matchLocale(
+            experienceTranslationOverrides?.override_language,
+            [locale]
+          );
+        }
```

Comment:
Rather than doing a more complex match here, let's just simplify this with some logic like: 
```suggestion
        if (experienceTranslationOverrides?.override_language) {
          // If translation overrides exist for this language, we will need to apply them below
          localeHasOverride = areLocalesEqual(experienceTranslationOverrides.override_language, locale);
        }
```

This would also allow us to not use the `matchLocale` helper which I think is overly complex for this.

Reply:
Moreover though: let's update the unit tests to cover these cases. Right now test coverage on `i18n-utils` is somewhere around 100% 👍

Reply:
This makes sense to me, thank you!